packages/terra-date-time-picker/src/DateTimePicker.jsx

```diff
@@ -318,9 +318,57 @@ class DateTimePicker extends React.Component {
 
   handleTimeChange(event, time) {
```

Comment:
This function is getting so big and complicated that it is hard to follow what it is doing. We should move some of this new logic to a separate function.

Reply:
We should probably just keep it this way for this iteration but it would help to add some code comments for each set of logic to make it easier to understand.

Reply:
Added some code comments for the keyboard shortcut section https://github.com/cerner/terra-framework/pull/1520/commits/e31d2250dc60b5bc379214c65e7312ff889cf584#diff-063daa4349be2bbad9c58209d280bfaee7a04b7e429163ebdb7784768f501a5cR326